storage/layerhandler.go

```diff
@@ -13,7 +13,9 @@ type LayerHandler interface {
 	// Layer if possible, or nil and an error when unsupported. This may
 	// directly serve the contents of the layer or issue a redirect to another
 	// URL hosting the content.
-	Resolve(layer Layer) (http.Handler, error)
+	// The LayerHandler must also accept an options map, although it may or may
+	// not use each provided option.
+	Resolve(layer Layer, options map[string]interface{}) (http.Handler, error)
```

Comment:
Oh yeah, we can definitely generate the URL inside the handler itself. I'll make this change and also improve the error handling in the scenario where they specify a layer handler and an incompatible underlying storage driver.